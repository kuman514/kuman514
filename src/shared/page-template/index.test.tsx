import { render, screen } from '@testing-library/react';

import PageTemplate from '.';

describe('PageTemplate', () => {
  it('should contain most recently modified date', async () => {
    render(
      <PageTemplate mostRecentlyEdited={new Date('2024-08-13')}>
        Lorem ipsum dolor sit amet.
      </PageTemplate>
    );

    expect(
      await screen.findByText(/최근 수정 일자: 2024년 8월 13일/i)
    ).not.toBeNull();
  });

  it('should match the snapshot', () => {
    const { container } = render(
      <PageTemplate
        customClassName="mt-24 text-6xl"
        mostRecentlyEdited={new Date('2024-08-13')}
      >
        Lorem ipsum dolor sit amet.
      </PageTemplate>
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
