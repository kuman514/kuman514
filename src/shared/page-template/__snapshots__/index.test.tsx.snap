// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageTemplate should match the snapshot 1`] = `
<main
  class="min-h-screen flex flex-col mt-24 text-6xl"
>
  <div
    class="w-full text-right"
  >
    최근 수정 일자: 
    2024
    년
     
    8
    월 
    13
    일
  </div>
  Lorem ipsum dolor sit amet.
</main>
`;
